### PR TITLE
fix: scarb checkout contracts for newer versions. Fix commit for current scarb commit

### PR DESCRIPTION
**Pull request by taco-paco**
…ent scarb release

### CODE REVIEW

File: plugin/src/features/Compilation/index.tsx [hunk 1]
```diff
@@ -749,6 +749,7 @@ const Compilation: React.FC<CompilationProps> = ({ setAccordian }) => {
       // const get all values of allFiles object
       const allFilesValues = Object.values(allFiles)
 
+      console.log('allFiles', allFiles)
```
Comment (satyambnsal) [inline plugin/src/features/Compilation/index.tsx:752]: Seems unnecessarily, perhaps just remove it.

File: plugin/src/features/Compilation/index.tsx [hunk 2]
```diff
@@ -779,10 +780,12 @@ const Compilation: React.FC<CompilationProps> = ({ setAccordian }) => {
               'toast',
               'Could not reach cairo compilation server'
             )
-            throw new Error('Cairo Compilation Request Failed')
+            throw new Error('Request to save code failed')
           }
         }
         const checkVal = Object.values(allFilesValues[i])[0]
+        console.log('checkVal', checkVal, allFilesValues[i], checkVal === '', checkVal === null, checkVal === undefined, isEmpty(checkVal))
```
Comment (satyambnsal) [inline plugin/src/features/Compilation/index.tsx:787]: Seems unnecessarily, perhaps just remove it.

File: plugin/src/utils/initial_scarb_codes.ts [hunk 3]
```diff
@@ -2,9 +2,16 @@ import axios from 'axios'
 
 export async function fetchGitHubFilesRecursively (
   repository: string,
-  path: string
+  path: string,
+  ref?: string
 ): Promise<Array<({ fileName: string, content: string, path: string } | null)>> {
-  const apiUrl = `https://api.github.com/repos/${repository}/contents/${path}`
+  const apiUrl = (function () {
```
Comment (satyambnsal) [inline plugin/src/utils/initial_scarb_codes.ts:8]: Curious why this needs to be IIFE ?
can't we just do this
```
const apiUrl  = `https://api.github.com/repos/${repository}/contents/${path}${ref === undefined ? '' : `?ref=${ref}`}`

**Comment by satyambnsal** [comment]
Overall looks good. Few minor comments.

File: plugin/src/features/Compilation/index.tsx (hunk 1 above)
Comment (taco-paco) [reply plugin/src/features/Compilation/index.tsx:752]: Oops, thought removed it. Thanks, good catch

File: plugin/src/utils/initial_scarb_codes.ts (hunk 3 above)
Comment (taco-paco) [reply plugin/src/utils/initial_scarb_codes.ts:8]: Just a preference. Can be anything. But I would leave it like this, other option looks a bit clumsy for me.

**Review by varex83 — APPROVED** [review]
LGTM

**Comment by varex83** [comment]
can you pull & merge develop into this branch please?

**Pull request merged**